designsafe/apps/workspace/api/views.py

```diff
@@ -777,6 +777,23 @@ def _submit_job(self, request, body, tapis, username):
             f"portalName: {settings.PORTAL_NAMESPACE}"
         ]
 
+        parameter_set = job_post.get("parameterSet", {})
+        env_variables = parameter_set.get("envVariables", [])
+        # Add user projects based on env var in the parameters.
+        projects_updated = False
+        for entry in env_variables:
+            if entry.get("key") == "_UserProjects":
+                projects = request.user.projects.order_by("-last_updated")
+                entry["value"] = " ".join(
+                    f"{project.uuid},{project.value['projectId'] if project.value['projectId'] != 'None' else project.uuid}"
+                    for project in projects[:settings.USER_PROJECTS_LIMIT]
+                )
+                projects_updated = True
+                break
+
+        if projects_updated:
+            job_post["parameterSet"]["envVariables"] = env_variables 
```

Comment:
Could this go right after line 790, and no need for `projects_updated` var?

Reply:
Yes, addressed this and tested again.